### ISSUE TITLE
spaced-comment evaluates shebang when shebang is only contents of file

### ISSUE THREAD

**Issue by kaicataldo** [eslint/eslint#7730]
**Tell us about your environment**

* **ESLint Version:** Master
* **Node Version:** v6.9.1
* **npm Version:** v3.10.9

**What parser (default, Babel-ESLint, etc.) are you using?**
Default
**Please show your full configuration:**

```
{
  "rules": {
    "spaced-comment": "error"
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

With the above configuration, the following does not warn:
```js
#!/usr/bin/env node
var foo;
```

while this does:
```js
#!/usr/bin/env node
```
`1:1  error  Expected space or tab after '//' in comment  spaced-comment`

**What did you expect to happen?**
For the behavior to be consistent (should always ignore shebang unless we add an option for this).

**What actually happened? Please include the actual, raw output from ESLint.**
Behavior was inconsistent. This is actually most likely a bug in how we handle shebang comments rather than a bug in the spaced-comment #rule itself. I'll make sure this works correctly as I continue to work on https://github.com/eslint/eslint/pull/7516.

**Comment by not-an-aardvark** [comment]
I can confirm that this is no longer an issue now that https://github.com/eslint/eslint/pull/7516 is merged.
